### PR TITLE
Internalize majority of cmd package

### PR DESCRIPTION
**Pull request by wagoodman**
This migrates the`cmd/syft/cli/commands` and `cmd/syft/cli/options` to `cmd/syft/internal`. In the process of doing so it became necessary to:
-  migrate integration tests to `cmd/syft/internal...` as well since there are specific `options.*` objects that help test against real CLI defaults (which is important in integration testing).
- add `internal/encoders` to act as an internal API for the full set of encoders that we can express, so that the CLI tests understand what formats should be under test.

### CODE REVIEW

File: cmd/syft/internal/test/integration/test-fixtures/.gitignore [hunk 1]
```diff
@@ -5,3 +5,6 @@
 # functionality), committing it seems like an acceptable exception.
 !image-pkg-coverage/pkgs/java/*.jar
 !image-pkg-coverage/pkgs/java/*.hpi
+
+**/go.sum
+!image-go-bin-arch-coverage/go.sum
```
Comment (spiffcs) [inline cmd/syft/internal/test/integration/test-fixtures/.gitignore:10]: +1 I see why this is needed now with the fixtures being moved to where they are

File: cmd/syft/internal/test/integration/test-fixtures/npm-lock/node_modules/collapse-white-space/package.json [hunk 2]
```diff
@@ -88,7 +88,7 @@
   },
   "scripts": {
     "build": "tsc",
-    "format": "remark . -qfo && prettier . -w --loglevel warn && xo --fix",
+    "format": "remark  -qfo && prettier  -w --loglevel warn && xo --fix",
```
Comment (spiffcs) [inline cmd/syft/internal/test/integration/test-fixtures/npm-lock/node_modules/collapse-white-space/package.json:91]: Was this a boyscout fix or removed by accident? Is the default for these just .?

**Review by spiffcs — APPROVED** [review]
Approve with just the nit on if a change was inadvertent from some find/replace command or if automated tooling did that

File: cmd/syft/internal/test/integration/test-fixtures/npm-lock/node_modules/collapse-white-space/package.json (hunk 2 above)
Comment (wagoodman) [reply cmd/syft/internal/test/integration/test-fixtures/npm-lock/node_modules/collapse-white-space/package.json:91]: weird! I have no idea what changed this, I'll change it back

File: internal/encoders/encoders.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	"github.com/anchore/syft/syft/format/spdxjson"
+	"github.com/anchore/syft/syft/format/spdxtagvalue"
+	"github.com/anchore/syft/syft/format/syftjson"
+	"github.com/anchore/syft/syft/format/table"
+	"github.com/anchore/syft/syft/format/template"
+	"github.com/anchore/syft/syft/format/text"
+	"github.com/anchore/syft/syft/sbom"
+)
+
+type Formats struct {
```
Comment (wagoodman) [inline internal/encoders/encoders.go:18]: this should be exported and generally usable for any consumer. I'll make that adjustment

File: cmd/syft/internal/test/integration/test-fixtures/npm-lock/node_modules/insert-css/example.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 var on = require('dom-event');
-var insertCss = require('./');
+var insertCss = require('insert-css');
```
Comment (wagoodman) [inline cmd/syft/internal/test/integration/test-fixtures/npm-lock/node_modules/insert-css/example.js:2]: I think intelliJ went a little sideways when refactoring (moving a dir)

**Pull request merged**